packages/client/src/declarations/index.ts

```diff
@@ -721,8 +736,35 @@ export async function getDeclarationsOfCurrentUser(): Promise<string> {
   const currentUserData = allUserData.find(
     (uData) => uData.userID === currentUserID
   )
-  const currentUserDeclarations: IDeclaration[] =
+  let currentUserDeclarations: IDeclaration[] =
     (currentUserData && currentUserData.declarations) || []
+
+  if (Roles.FIELD_AGENT.includes(currentUserRole) && currentUserData) {
+    currentUserDeclarations = currentUserData.declarations.filter((d) => {
+      if (d.downloadStatus === DOWNLOAD_STATUS.DOWNLOADED) {
+        const history = d.originalData?.history as unknown as IDynamicValues
```

Comment:
Would this be more suitable?

```suggestion
        const history = d.originalData?.history as unknown as History[]
```

```ts
import { History } from '@client/utils/gateway'
```